### PR TITLE
Feature/add combine example

### PR DESCRIPTION
**Pull request by andreshbarrueco**

### CODE REVIEW

File: CiMAExample/Shared/Domain/Repositories/HttpRepository.swift [hunk 1]
```diff
@@ -1,12 +1,12 @@
 //
 //  HttpRepository.swift
 //  CiMAExample (iOS)
-//
```
Comment (babel-cdm) [inline CiMAExample/Shared/Domain/Repositories/HttpRepository.swift:4]: why do you delete this?

File: CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinator.swift [hunk 2]
```diff
@@ -1,16 +1,13 @@
 //
 //  MainCoordinator.swift
 //  CiMAExample (iOS)
-//
```
Comment (babel-cdm) [inline CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinator.swift:4]: why do you delete this?

File: CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinatorView.swift [hunk 3]
```diff
@@ -1,9 +1,6 @@
 //
 //  MainCoordinatorView.swift
 //  CiMAExample (iOS)
-//
```
Comment (babel-cdm) [inline CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinatorView.swift:4]: why do you delete this?

File: CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinator.swift [hunk 4]
```diff
@@ -30,7 +27,7 @@ public class MainCoordinator: BaseCoordinator, MainCoordinatorProtocol {
     }
     
     public func navigateToForm(delegate: FormDelegate) {
-        formViewModel = DependencyInjector.shared.getFormViewModel(coordinator: self,
+        formViewModel = DependencyInjector.shared.getFormViewModel(coordiantor: self,
```
Comment (babel-cdm) [inline CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinator.swift]: it was well written, you put it wrong

File: CiMAExample/Shared/Presentation/Common/Dependencies/DependencyInjector.swift [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import Foundation
+
+final class DependencyInjector {
+    static let shared = DependencyInjector()
+    
+    func getMainViewModel( coordinator: MainCoordinator ) -> MainViewModel {
+        .init(coordinator: coordinator)
+    }
+    
+    func getPokemonListViewModel( coordinator: MainCoordinator, isCompletionRequestSelected: Bool) -> PokemonListViewModel {
```
Comment (babel-cdm) [inline CiMAExample/Shared/Presentation/Common/Dependencies/DependencyInjector.swift:17]: Remove the space before `coodinator`

File: CiMAExample/Shared/Presentation/Common/Dependencies/DependencyInjector.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+//
+//  Created by Andres Heras Barrueco on 22/9/22.
+//
+
+import Foundation
+
+final class DependencyInjector {
+    static let shared = DependencyInjector()
+    
+    func getMainViewModel( coordinator: MainCoordinator ) -> MainViewModel {
```
Comment (babel-cdm) [inline CiMAExample/Shared/Presentation/Common/Dependencies/DependencyInjector.swift]: Remove the space before `coodinator`

File: CiMAExample/Shared/Presentation/Common/Dependencies/DependencyInjector.swift [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        .init(coordinator: coordinator)
+    }
+    
+    func getPokemonListViewModel( coordinator: MainCoordinator, isCompletionRequestSelected: Bool) -> PokemonListViewModel {
+        return PokemonListViewModel(coordinator: coordinator, pokemonListUseCase: getPokemonListUseCase(), isCompletionRequestSelected: isCompletionRequestSelected)
+    }
+    
+    func getFormViewModel(coordiantor: MainCoordinator, delegate: FormDelegate? = nil) -> FormViewModel{
```
Comment (babel-cdm) [inline CiMAExample/Shared/Presentation/Common/Dependencies/DependencyInjector.swift]: `coordinator`

File: CiMAExample/Shared/Presentation/Modules/Form/FormViewModel.swift [hunk 8]
```diff
@@ -39,7 +38,7 @@ final class FormViewModel: BaseViewModel<MainCoordinatorProtocol> {
 extension FormViewModel {
     static var sample: FormViewModel {
         let coordinator = MainCoordinator.sample
-        let viewModel = DependencyInjector.shared.getFormViewModel(coordinator: coordinator,
+        let viewModel = DependencyInjector.shared.getFormViewModel(coordiantor: coordinator,
```
Comment (babel-cdm) [inline CiMAExample/Shared/Presentation/Modules/Form/FormViewModel.swift]: same as before

**Review by babel-cdm — CHANGES_REQUESTED** [review]

File: CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinator.swift (hunk 4 above)
Comment (andreshbarrueco) [reply CiMAExample/Shared/Presentation/Common/Coordinators/MainCoordinator/MainCoordinator.swift]: dislexia ^^

**Pull request closed without merging**